opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/RuntimeEnvironment.java

```diff
@@ -497,20 +535,8 @@ public boolean hasGroups() {
      * @param groups the set of groups to use
      */
     public void setGroups(Set<Group> groups) {
-        populateGroups(groups, new TreeSet<Project>(getProjects().values()));
-        threadConfig.get().setGroups(groups);
-    }
-
-    /**
-     * Register this thread in the thread/configuration map (so that all
-     * subsequent calls to the RuntimeEnvironment from this thread will use the
-     * same configuration
-     *
-     * @return this instance
-     */
-    public RuntimeEnvironment register() {
-        threadConfig.set(configuration);
-        return this;
+        populateGroups(groups, new TreeSet<>(getProjects().values()));
+        setConfigurationValue("groups", groups);
     }
 
     /**
```

Comment:
repository_map doesn't need locking?

Reply:
it initialized as `ConcurrentHashMap` so my guess is no. Also, it is defined within `RuntimeEnvironment`, not `Configuration`.